### PR TITLE
flow 3: Flow updates to constants-dev

### PR DESCRIPTION
**Pull request by chrisnojima**

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers rebase of part 3 of the flow fixes

File: shared/constants/images.js [hunk 1]
```diff
@@ -1,8 +1,15 @@
+// @flow
 export const services = {
+  // $FlowIssue
```
Comment (chromakode) [inline shared/constants/images.js]: What's going on here? Could you please elaborate in the comments?

File: shared/dev/component-sheet.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
+// @flow
 import React, {Component} from 'react'
+// $FlowIssue
```
Comment (chromakode) [inline shared/dev/component-sheet.js]: What's the flow issue here?

File: shared/devices/existing-device/index.js [hunk 3]
```diff
@@ -33,8 +34,14 @@ class ExistingDevice extends Component<void, Props, void> {
 
     return (
       <Render
-        onSubmitComputer={() => this.props.onSubmit(otherDeviceComputer)}
-        onSubmitPhone={() => this.props.onSubmit(otherDevicePhone)}
+        onSubmitComputer={() => {
+          // $FlowIssue
```
Comment (chromakode) [inline shared/devices/existing-device/index.js]: Can you please comment what the flow issue is here?

**Comment by chromakode** [comment]
Changes look good. I think we should put explanations with `$FlowIssue` because it can be pretty opaque when read out of context. Also, it looks like there are a few uncommitted autogenerated protocol changes that are causing the Jenkins build to fail: https://ci.keybase.io/job/client/job/PR-3779/2/execution/node/107/log/

File: shared/dev/component-sheet.js (hunk 2 above)
Comment (chrisnojima) [inline shared/dev/component-sheet.js]: i've entirely removed this file

File: shared/devices/existing-device/index.js [hunk 4]
```diff
@@ -14,22 +15,11 @@ type Props = {
 
 class ExistingDevice extends Component<void, Props, void> {
   render () {
-    let otherDeviceComputer = null
-    let otherDevicePhone = null
+    const otherRoleIsNew = this.props.myDeviceRole === codePageDeviceRoleExistingPhone ||
```
Comment (chrisnojima) [inline shared/devices/existing-device/index.js]: This refactor was required cause flow was confused about the state of the variables. it complained that they might be null but the switch covered all cases and had a default so i changed it to a different for so it'd not complain

@chromakode (wanted comments on the original $FlowIssue, but he's on vacation)

**Comment by chrisnojima** [comment]
@keybase/react-hackers can i get 👀  since @chromakode is out and i've responded to his feedback

**Comment by cjb** [comment]
all looks good :thumbsup:

**Pull request merged**